modules/custom/openy_focal_point/src/Form/OpenYFocalPointCropForm.php

```diff
@@ -0,0 +1,218 @@
+        '#crop_preview_image_style' => 'crop_thumbnail',
+        '#show_default_crop' => FALSE,
+        '#show_crop_area' => TRUE,
+        '#warn_multiple_usages' => FALSE,
+        '#crop_types_required' => [$crop_type],
+      ];
+    }
+
+    $form['submit'] = [
+      '#type' => 'submit',
+      '#value' => $this->t('Save Crop'),
+      '#ajax' => [
+        'callback' => '::ajaxSave',
+      ],
+    ];
+
+    $form['remove'] = [
+      '#type' => 'submit',
+      '#value' => $this->t('Remove cropping (use default focal point)'),
+      '#ajax' => [
+        'callback' => '::ajaxRemoveCropping',
+      ],
+    ];
+
+    $form['close_dialog'] = [
+      '#type' => 'submit',
+      '#value' => $this->t('Close'),
+      '#ajax' => [
+        'callback' => '::closePopup',
+      ],
+    ];
+
+    $form['#attached']['library'][] = 'openy_focal_point/openy_focal_point';
+
+    return $form;
+  }
+
+  public static function ajaxSave(array $form, FormStateInterface $form_state) {
+    /** @var \Drupal\file\Entity\File $file */
+    $file = $form_state->get('file');
+
+    $crop_type = $form_state->get('crop_type');
+
+    $input = $form_state->getUserInput();
+    $crop_properties = $input['prgf_teaser']['crop_wrapper'][$crop_type]['crop_container']['values'];
+    $x = (int) ($crop_properties['x'] + $crop_properties['width'] / 2);
+    $y = (int) ($crop_properties['y'] + $crop_properties['height'] / 2);
+
+    $crop = Crop::findCrop($file->getFileUri(), $crop_type);
+    if ($crop) {
+      if ($crop_properties['height'] == 0 && $crop_properties['width'] == 0) {
+        $crop->delete();
+        $crop = NULL;
+      }
+      else {
+        $crop->setSize($crop_properties['width'], $crop_properties['height']);
+        $crop->setPosition($x, $y);
+      }
+    }
+    else {
+      $crop_storage = \Drupal::entityTypeManager()->getStorage('crop');
+      $crop = $crop_storage->create([
+        'type' => $crop_type,
+        'entity_id' => $file->id(),
+        'entity_type' => 'file',
+        'uri' => $file->getFileUri(),
+        'x' => $x,
+        'y' => $y,
+        'width' => $crop_properties['width'],
+        'height' => $crop_properties['height'],
+      ]);
+    }
+
+    if ($crop) {
+      $crop->save();
+    }
+
+    $image = \Drupal::service('image.factory')->get($file->getFileUri());
+    image_path_flush($image->getSource());
+
+    $ajax = new AjaxResponse();
+    $ajax->addCommand(new RerenderThumbnailCommand('.focal-point-derivative-preview-image'));
+    return $ajax;
+  }
+
+  public static function ajaxRemoveCropping(array $form, FormStateInterface $form_state) {
+    /** @var \Drupal\file\Entity\File $file */
+    $file = $form_state->get('file');
+
+    $crop_type = $form_state->get('crop_type');
+
+    $crop = Crop::findCrop($file->getFileUri(), $crop_type);
+    if ($crop) {
+        $crop->delete();
+    }
+
+    $image = \Drupal::service('image.factory')->get($file->getFileUri());
```

Comment:
we should pass service via DI

Reply:
@podarok we use this service in public static method, so we don't have instance of the form. This is why we can not use DI here.